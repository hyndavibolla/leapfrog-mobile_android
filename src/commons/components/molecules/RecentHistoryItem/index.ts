export { default as RecentHistoryItem } from './RecentHistoryItem';
