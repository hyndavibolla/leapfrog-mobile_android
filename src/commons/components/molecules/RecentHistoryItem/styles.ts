import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE } from '_constants';

export const styles = {
  itemSearchHistory: css`
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-top: 16px;
  `,
  textSearchHistory: css`
    color: ${COLOR.BLACK};
    font-family: ${FONT_FAMILY.MEDIUM};
    font-size: ${FONT_SIZE.SMALLER};
    margin-left: 16px;
  `
};
