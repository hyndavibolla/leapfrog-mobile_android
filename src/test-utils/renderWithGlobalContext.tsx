import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
