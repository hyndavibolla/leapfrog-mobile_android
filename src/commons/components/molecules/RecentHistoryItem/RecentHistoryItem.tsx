import React, { memo } from 'react';
import { TouchableOpacity, Text } from 'react-native';

import { Icon } from '../../atoms/Icon';

import { useTestingHelper } from '_utils/useTestingHelper';
import { ICON, FONT_SIZE, COLOR } from '_constants';
import { styles } from './styles';

export interface Props {
  title: string;
  category: boolean;
  onPress: () => void;
}

export const RecentHistoryItem = ({ title, category, onPress }: Props) => {
  const { getTestIdProps } = useTestingHelper('recent-search');

  return (
    <TouchableOpacity {...getTestIdProps('item')} style={styles.itemSearchHistory} onPress={onPress}>
      <Icon name={!category ? ICON.SEARCH : ICON.FILTERS} size={FONT_SIZE.PETITE} color={COLOR.BLACK} />
      <Text style={styles.textSearchHistory}>{title}</Text>
    </TouchableOpacity>
  );
};

export default memo(RecentHistoryItem);
