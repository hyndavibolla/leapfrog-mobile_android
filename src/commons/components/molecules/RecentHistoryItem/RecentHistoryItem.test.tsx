import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import RecentHistoryItem, { Props } from './RecentHistoryItem';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';

describe('RecentHistoryItem', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      title: 'title',
      onPress: jest.fn(),
      category: false
    };
  });

  it('should render', () => {
    const { toJSON } = renderWithGlobalContext(<RecentHistoryItem {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with identifier', () => {
    props.category = true;
    const { getByTestId } = renderWithGlobalContext(<RecentHistoryItem {...props} />);
    expect(getByTestId('recent-search-item')).toBeTruthy();
  });

  it('should execute onPress', async () => {
    const { getByTestId } = renderWithGlobalContext(<RecentHistoryItem {...props} />);
    fireEvent.press(getByTestId('recent-search-item'));
    expect(props.onPress).toBeCalled();
  });
});
