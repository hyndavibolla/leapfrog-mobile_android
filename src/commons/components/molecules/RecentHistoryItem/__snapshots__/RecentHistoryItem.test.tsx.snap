// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentHistoryItem should render 1`] = `
<View
  accessibilityLabel="recent-search-item"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "marginTop": 16,
      "opacity": 1,
    }
  }
  testID="recent-search-item"
>
  <Text
    accessibilityLabel="icon-text"
    accessible={true}
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontSize": 14,
        },
        undefined,
        Object {
          "fontFamily": "sywmax",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="icon-text"
  >
    î¤´
  </Text>
  <Text
    style={
      Object {
        "color": "#262C2F",
        "fontFamily": "SFProDisplay-Medium",
        "fontSize": 16,
        "marginLeft": 16,
      }
    }
  >
    title
  </Text>
</View>
`;
